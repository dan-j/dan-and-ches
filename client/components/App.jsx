import React from 'react';
import Header from './Header';
import ContentContainer from '../containers/ContentContainer';
import Footer from './Footer';

const App = (props) => {
  const { header } = props.routes[props.routes.length - 1];
  return (
    <div>
      <Header {...header} />
      <ContentContainer>
        {props.children}
      </ContentContainer>
      <Footer />
    </div>
  );
};

App.propTypes = {
  children: React.PropTypes.node,
  routes: React.PropTypes.arrayOf(
    React.PropTypes.shape({
      header: React.PropTypes.shape({
        preHeader: React.PropTypes.string,
        mainHeader: React.PropTypes.string.isRequired,
      }),
    }),
  ),
};

export default App;
