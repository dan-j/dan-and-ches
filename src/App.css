@media only screen and (max-width: 767px) {
  h1 {
    font-size: 3em;
    letter-spacing: 3px;
  }

  h2 {
    font-size: 2.1em;
    letter-spacing: 3px;
  }

  h3 {
    font-size: 1.6em;
    letter-spacing: 3px;
  }
}

@media only screen and (min-width: 768px) {
  h1 {
    font-size: 4em;
    letter-spacing: 4px;
  }

  h2 {
    font-size: 3em;
    letter-spacing: 3px;
  }

  h3 {
    font-size: 2em;
    letter-spacing: 3px;
  }
}