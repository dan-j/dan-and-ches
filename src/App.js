import * as React from 'react';
import {Component} from 'react';
import {Grid} from 'react-bootstrap';
import './App.css';
import Header from './Header';
import Navigation from './Navigation';
import Content from './Content';

export default class App extends Component {
  render() {
    return (
          <Grid>
            <Header/>
            <Navigation/>
            <Content/>
          </Grid>
    );
  }
}