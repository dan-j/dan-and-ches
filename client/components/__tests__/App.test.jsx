import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

it('App renders as expected', () => {
  const app = renderer.create(<App />);
  const tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});
