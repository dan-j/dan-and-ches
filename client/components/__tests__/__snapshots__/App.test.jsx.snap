exports[`test App renders as expected 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <button
    onClick={[Function]}>
    Click me!
  </button>
  <div>
    Not Loaded
  </div>
</div>
`;
